### DISCUSSION TITLE
float16 in StreamPeer

### DISCUSSION THREAD

**Discussion by theraot** [godotengine/godot-proposals#5983] [Engine Core]
Currently in [StreamPeer](https://docs.godotengine.org/en/stable/classes/class_streampeer.html) I can put and get float and double, and I can put and get int16, but no half precision floating point (float16).

I want to build-in methods to put and get float16 for network traffic and serialization optimization.

Converting GDScript `float` to float16 stored in an `int` to use with `get_int16` and `put_int16` would be error prone and less efficient.

Furthermore, I think the float and double methods could use a rename. So:

- `float get_float16()`, `void put_float16(float value)` (currently not existent)
- `float get_float32()`, `void put_float32(float value)` (currently `float get_float()`, `void put_float(float value)`)
- `float get_float64()`, `void put_float64(float value)` (currently `float get_double()`, `void put_double(float value)`)

*Notice the mix of language here: the API currently uses the words `double` and `float` instead of `double` and `single`. Meanwhile the methods working with `int` say the size in bits.*

**Comment by Calinou** [comment]
I'd suggest `get_half()` and `put_half()` instead of renaming the existing methods.

**Comment by theraot** [comment]
Naming aside the feature is already in the engine. I'm closing this.
